agent/dns.go

```diff
@@ -505,18 +505,23 @@ func (d *DNSServer) handleQuery(resp dns.ResponseWriter, req *dns.Msg) {
 	}
 }
 
-func (d *DNSServer) soa(cfg *dnsConfig) *dns.SOA {
+func (d *DNSServer) soa(cfg *dnsConfig, questionName string) *dns.SOA {
+	domain := d.domain
+	if strings.HasSuffix(questionName, d.altDomain) {
+		domain = d.altDomain
+	}
```

Comment:
A check if `d.altDomain` is not empty need to be done before, otherwise this will always be true.
Also it's safer if we do the check with a `.` prefix
```suggestion
	if d.altDomain !="" && strings.HasSuffix(questionName, "." + d.altDomain) {
		domain = d.altDomain
	}
```